### PR TITLE
Remove core dependency on viz

### LINKED ISSUE #5589: Refactor BuildResource and BuildResource to remove public-api dependency on viz
Refactor the `BuildResource` and `BuildResources` functions in `controller/api/util/api_utils.go` that are used by `cli/cmd/metrics` so that instead of returning viz' `pb.Resource` they return an internal struct or something from k8s ' apimachinery. That way we can remove the public-api dependency on viz.

Other places in viz that depend on these functions (`stat`, `edges`, `routes` and `tap` (soon)) would need to convert between `pb.Resource` and that new struct, if needed.

### PR DESCRIPTION
**Pull request by adleong**
Fixes #5589 

The core control plane has a dependency on the viz package in order to use the `BuildResource` function.  This "backwards" dependency means that the viz source code needs to be included in core docker-builds and is bad for code hygiene.

We move the `BuildResource` function into the viz package.  In `cli/cmd/metrics.go` we replace a call to `BuildResource` with a call directly to `CanonicalResourceNameFromFriendlyName`.

Signed-off-by: Alex Leong <alex@buoyant.io>

### CODE REVIEW

File: cli/cmd/metrics.go [hunk 1]
```diff
@@ -122,18 +122,25 @@ func getPodsFor(ctx context.Context, clientset kubernetes.Interface, namespace s
 	// objects in Public API protobuf form for submission to the Public API
```
Comment (Pothulapati) [inline cli/cmd/metrics.go:122]: Maybe remove/update the TODO as it talks about `BuildResource` which we are not using anymore?

File: viz/pkg/util.go [hunk 2]
```diff
@@ -1,6 +1,17 @@
 package pkg
```
Comment (Pothulapati) [inline viz/pkg/util.go:1]: nit, TIOLI: I see we were already doing this with `util.go`. Maybe move this into a `utils` pkg. `package pkg` feels a bit confusing.

**Review by Pothulapati — APPROVED** [review]
LGTM! :shipit:

**Review by Pothulapati — APPROVED** [review]

**Review by alpeb — APPROVED** [review]
LGTM, module @Pothulapati remark about that TO-DO :+1:

**Review by kleimkuhler — APPROVED** [review]
Yep aside from comments already made looks good to me 👍

**Pull request merged**